control-plane/tenancy/namespace/namespace_controller.go

```diff
@@ -0,0 +1,138 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package namespace
+
+import (
+	"context"
+	"fmt"
+
+	mapset "github.com/deckarep/golang-set"
+	"github.com/go-logr/logr"
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/hashicorp/consul-k8s/control-plane/connect-inject/common"
+	"github.com/hashicorp/consul-k8s/control-plane/connect-inject/constants"
+	"github.com/hashicorp/consul-k8s/control-plane/consul"
+	"github.com/hashicorp/consul-k8s/control-plane/namespaces"
+)
+
+type Controller struct {
+	client.Client
+
+	// ConsulClientConfig is the config for the Consul API client.
+	ConsulClientConfig *consul.Config // TODO: remove
+
+	// ConsulServerConnMgr is the watcher for the Consul server addresses.
+	ConsulServerConnMgr consul.ServerConnectionManager
+
+	// AllowK8sNamespacesSet determines kube namespace that are reconciled.
+	AllowK8sNamespacesSet mapset.Set
+
+	// DenyK8sNamespacesSet determines kube namespace that are ignored.
+	DenyK8sNamespacesSet mapset.Set
+
+	// Partition is not required. It should already be set in the API ClientConfig - really?
+
+	// ConsulDestinationNamespace is the name of the Consul namespace to create
+	// all config entries in. If EnableNSMirroring is true this is ignored.
+	ConsulDestinationNamespace string
+
+	// MirroringK8s causes Consul namespaces to be created to match the
+	// k8s namespace of any config entry custom resource. Config entries will
+	// be created in the matching Consul namespace.
+	MirroringK8s bool
+
+	// MirroringK8sPrefix is an optional prefix that can be added to the Consul
+	// namespaces created while mirroring. For example, if it is set to "k8s-",
+	// then the k8s `default` namespace will be mirrored in Consul's
+	// `k8s-default` namespace.
+	MirroringK8sPrefix string
```

Comment:
I could be wrong, but i thought in v2 tenancy we are going to map the Kubernetes namespace to a Consul namespace with the same name and not support destination namespaces and prefixes? as in, always be mirroring?